plugins/logs/plugin.go

```diff
@@ -277,6 +278,21 @@ func (c *Config) validateAndInjectDefaults(services []string, plugins []string)
 		return fmt.Errorf("invalid decision_log config, must have a `service`, `plugin`, or `console` logging enabled")
 	}
 
+	if c.Reporting.Trigger == nil {
+		c.Reporting.Trigger = trigger
+	} else {
+		if *c.Reporting.Trigger != *trigger {
+			return fmt.Errorf("invalid decision_log config, trigger mode mismatch")
+		}
+	}
```

Comment:
💭 I guess we could have the validation `switch` below happen earlier -- so typos will be reported as "invalid trigger mode", not as a "mismatch", if both discovery and logs have the trigger mode configured...?

Reply:
Updated the validation logic to move up the `switch` case in all plugins.